### PR TITLE
Add circleci configuration

### PR DESCRIPTION
**Pull request by BravoSimone**
I'm adding CircleCI on this project and making the test suite green.

### CODE REVIEW

**Comment by BravoSimone** [comment]
@kennyadsl since we're removing the old badges we should also add the new CircleCI badge. Unfortunately I can't generate it because I don't have permissions to edit CircleCI's settings.

**Comment by kennyadsl** [comment]
Let's use this one:

[CircleCI](https://circleci.com/gh/solidusio-contrib/solidus_searchkick)

**Comment by BravoSimone** [comment]
Done, do we want to merge the new executors in the orb first or are we good with merging the development preview?

**Comment by kennyadsl** [comment]
Orbs development versions last ~30 days so I think we should merge new executors first

File: Gemfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 source 'https://rubygems.org'
 
-gem 'solidus', '~> 1.4.0'
+gem 'solidus', github: 'solidusio/solidus', branch: 'master'
```
Comment (kennyadsl) [inline Gemfile]: we should change the solidus version based on an env variable, see [here](https://github.com/solidusio/solidus_auth_devise/blob/master/Gemfile) for an example.

File: README.md [hunk 2]
```diff
@@ -1,6 +1,5 @@
-[![Build Status](https://travis-ci.org/elevatorup/solidus_searchkick.svg?branch=master)](https://travis-ci.org/elevatorup/solidus_searchkick)
+[![CircleCI](https://circleci.com/gh/solidusio-contrib/solidus_searchkick.svg?style=svg)](https://circleci.com/gh/solidusio-contrib/solidus_searchkick)
 [![Code Climate](https://codeclimate.com/github/elevatorup/solidus_searchkick/badges/gpa.svg)](https://codeclimate.com/github/elevatorup/solidus_searchkick)
```
Comment (kennyadsl) [inline README.md]: Does codeclimate use the coverage stuff you removed later?

Comment (BravoSimone) [reply README.md]: You're right. I'm removing also this right now and I'll open an issue to discuss if and what we want to add as a code analyzer.

**Pull request merged**